src/mongocxx/test/collection.cpp

```diff
@@ -42,6 +42,36 @@ TEST_CASE("CRUD functionality", "[driver::collection]") {
         REQUIRE(i == 1);
     }
 
+    SECTION("insert and read multiple documents", "[collection]") {
+        bsoncxx::builder::stream::document b1;
+        bsoncxx::builder::stream::document b2;
+        bsoncxx::builder::stream::document b3;
+
+        b1 << "_id" << bsoncxx::oid{bsoncxx::oid::init_tag} << "x" << 1;
```

Comment:
Can you please add some tests for objects without _id fields?  Then we'll hit the problematic path above in the builder.